js/slidesModel.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * The base class for Slides.
+ */
+class BaseSlide {
+    constructor(label) {
+        if (!BaseSlide._id) { BaseSlide._id = 0; }
+        BaseSlide._id++;
+        this._uid = `${this.constructor.name}-${BaseSlide._id}`;
+        this._label = label;
```

Comment:
Label can be an object that has a category, and an optional ordering. ie {category: "Diffususion", order:1}.